### PR TITLE
bump-formula-pr: add --message option

### PR DESCRIPTION
**Pull request by zmwangx**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Add a `--message=<message>` option to `bump-formula-pr` so that user can supply a piece of message to be included in the PR message.

The motivation comes from automatically bumping `imagemagick`. I have a script to do it, but I want to include a reminder to maintainers:

> Please remember to
> ```
> brew mirror imagemagick
> ```
> when merging.

Since `brew-bump-formula-pr` doesn't allow custom PR message, I have to copy the message to clipboard at the end of the script, and remember to paste that into a comment when the PR has been created. I'd like to be able to do

```
brew bump-formula-pr --url=https://dl.bintray.com/homebrew/mirror/imagemagick-6.0.0-0.tar.xz --mirror=https://www.imagemagick.org/download/ImageMagick-6.0.0-0.tar.xz --sha256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 --message=$'Please remember to\n```\nbrew mirror imagemagick\n```\nwhen merging.' imagemagick
```

and forget about it.

I understand that this is a niche feature, but `bump-formula-pr` is a very niche dev cmd to begin with, so I figured it doesn't hurt to suggest this.

### CODE REVIEW

**Comment by zmwangx** [comment]
Demo in https://github.com/Homebrew/homebrew-core/pull/6939, created with

```
brew bump-formula-pr --url=https://dl.bintray.com/homebrew/mirror/imagemagick-10.0.0-0.tar.xz --mirror=https://www.imagemagick.org/download/ImageMagick-10.0.0-0.tar.xz --sha256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 --message=$'Please remember to\n```\nbrew mirror imagemagick\n```\nwhen merging.' imagemagick
```

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 1]
```diff
@@ -259,8 +262,10 @@ def bump_formula_pr
         remote = Utils.popen_read("hub fork 2>&1")[/fatal: remote (.+) already exists\./, 1]
         odie "cannot get remote from 'hub'!" if remote.to_s.empty?
         safe_system "git", "push", "--set-upstream", remote, "#{branch}:#{branch}"
-        safe_system "hub", "pull-request", "--browse", "-m",
-          "#{formula.name} #{new_formula_version}#{devel_message}\n\nCreated with `brew bump-formula-pr`."
+        pr_message = "#{formula.name} #{new_formula_version}#{devel_message}\n\nCreated with `brew bump-formula-pr`."
+        user_message = ARGV.value('message')
+        pr_message += "\n\n---\n\n#{user_message}" if user_message
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: Use `EOS.undent` here to make it more obvious how the newlines fit in. I realise `\n` is used above but it'd be nice to improve that too.

**Comment by MikeMcQuaid** [comment]
I'm 🆒 with this option 👍

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 2]
```diff
@@ -259,8 +262,10 @@ def bump_formula_pr
         remote = Utils.popen_read("hub fork 2>&1")[/fatal: remote (.+) already exists\./, 1]
         odie "cannot get remote from 'hub'!" if remote.to_s.empty?
         safe_system "git", "push", "--set-upstream", remote, "#{branch}:#{branch}"
-        safe_system "hub", "pull-request", "--browse", "-m",
-          "#{formula.name} #{new_formula_version}#{devel_message}\n\nCreated with `brew bump-formula-pr`."
+        pr_message = "#{formula.name} #{new_formula_version}#{devel_message}\n\nCreated with `brew bump-formula-pr`."
+        user_message = ARGV.value('message')
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: Double-quoted strings: https://bot.brew.sh/job/Brew/2946/version=any_sierra/testReport/junit/brew-test-bot/sierra/style/

**Comment by zmwangx** [comment]
@MikeMcQuaid Done

**Comment by MikeMcQuaid** [comment]
Thanks @zmwangx!

**Pull request merged**